app/main.py

```diff
@@ -2,0 +1,86 @@
+class Animal:
+    """
+    Class which contains animals. It stores list of in animals list.
+    """
+    animals = []
+
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: type(bool) = True) -> None:
+        """
+        Create instance of Animal.
+        :param name: animal's name.
+        :param is_hungry: boolean that shows if animal is ready to eat
+        with True default value.
+        :param appetite: integer that shows how much food points.
+        this animal need to eat to be full.
+
+        """
+        self.name = name
+        self.is_hungry = is_hungry
+        self.appetite = appetite
+        Animal.animals.append(self)
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        """
+        Function that feed animal if it's is_hungry equal True.
+        It prints how many food points animal ate, return
+        value of animal's appetite and change is_hungry to False.
+        If animal is_hungry is equal to False, function return 0.
+        :return:
+        """
+        if self.is_hungry is True:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        elif self.is_hungry is False:
+            return 0
+        elif self.is_hungry is None:
+            return 0
+        else:
+            return 0
+
+
+class Cat(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3)
+        self.is_hungry = is_hungry
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name : str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7)
+        self.is_hungry = is_hungry
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals_list: list[Animal.animals]) -> int:
```

Comment:
The type hint `list[Animal.animals]` is incorrect. It should be `list[Animal]` to indicate a list of `Animal` instances.